rkt/api_service.go

```diff
@@ -735,6 +735,12 @@ func (s *v1AlphaAPIServer) GetLogs(request *v1alpha.GetLogsRequest, server v1alp
 		stage1Path = fmt.Sprintf("/overlay/%s/upper/", stage1TreeStoreID)
 	}
 	path := filepath.Join(getDataDir(), "/pods/run/", request.PodId, stage1Path, "/var/log/journal/")
+
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		// in case no journal is present, simply send an empty logs response
+		return server.Send(&v1alpha.GetLogsResponse{})
+	}
```

Comment:
If we don't send a proper error and `request.Follow` is true, the client will wait forever but the server will never send more data.